v1.25.1 -- Resolve https://github.com/Bioconductor/BiocCheck/issues/57

lcolladotor opened the pull request:
Hi,

This pull request resolves https://github.com/Bioconductor/BiocCheck/issues/57 by updating the `handleMessage()` calls after the formatting check fails. Instead of mentioning `formatR`, these messages now refer to:

* `styler`,
* `biocthis`,
* the `BiocCheck` vignette.

Furthermore, the `BiocCheck` vignette section on formatting still mentions `formatR`, but it now also mentions in more detail:

* `styler`: how to install and run it
* `biocthis`: as a companion to `styler` with code how to run it (but not install it since it's optional and the install instructions will hopefully change soon once this package is submitted/reviewed).
* a quick overview of the differences between `formatR` and `styler`, though this is mostly done by highlighting current limitations of `formatR`.
* how RStudio Desktop's `Reformat code` button (under the magic wand) can help break long lines which is something that `styler` cannot do as discussed in more detail at
https://github.com/r-lib/styler/issues/247. `formatR` can do this, but personally, the fact that it broke valid R code in some of my packages made me trust its output less.

The `NEWS` and `DESCRIPTION` files have been updated accordingly.

Best,
Leo

lcolladotor committed:
v1.25.1 -- Resolve https://github.com/Bioconductor/BiocCheck/issues/57

This commit resolves https://github.com/Bioconductor/BiocCheck/issues/57
by updating the `handleMessage()` calls after the formatting check
fails. Instead of mentioning `formatR`, these messages now refer to:

* `styler`,
* `biocthis`,
* the `BiocCheck` vignette.

Furthermore, the `BiocCheck` vignette section on formatting still
mentions `formatR`, but it now also mentions in more detail:

* `styler`: how to install and run it
* `biocthis`: as a companion to `styler` with code how to run it
(but not install it since it's optional and the install instructions
will hopefully change soon once this package is submitted/reviewed).
* a quick overview of the differences between `formatR` and `styler`,
though this is mostly done by highlighting current limitations of
`formatR`.
* how RStudio Desktop's `Reformat code` button (under the magic wand)
can help break long lines which is something that `styler` cannot do
as discussed in more detail at
https://github.com/r-lib/styler/issues/247. `formatR` can do this,
but personally, the fact that it broke valid R code in some of my
packages made me trust its output less.

The NEWS and DESCRIPTION files have been updated accordingly.

mtmorgan commented on R/checks.R:
```diff
@@ -1431,7 +1431,11 @@ checkFormatting <- function(pkgdir, nlines=6)
             "See http://bioconductor.org/developers/how-to/coding-style/")
         handleMessage(
-            "See FormatR package: https://cran.r-project.org/web/packages/formatR/index.html")
+            "See styler package: https://styler.r-lib.org/ ",
```
please point to the CRAN repository

mtmorgan commented on R/checks.R:
```diff
@@ -1431,7 +1431,11 @@ checkFormatting <- function(pkgdir, nlines=6)
+            "as described in the BiocCheck vignette.")
+        handleMessage(
+            "See also biocthis: ",
+            "https://lcolladotor.github.io/biocthis/reference/bioc_style.html")
```
we can't reference a package that is not in CRAN or Biocondcutor

mtmorgan commented on vignettes/BiocCheck.Rmd:
```diff
@@ -550,10 +550,48 @@ The output of this check includes the first 6 offending lines of code;
+## Automatically re-format the R code in your package
+styler::style_pkg(transformers = styler::tidyverse_style(indent_by = 4))
+
+## If using biocthis:
```
This too is problematic, with biocthis not yet in Bioconductor. I also feel that, even if biocthis were in Bioconductor, it seems likely that we would like for it to have been tested and used 'for a while' before sending users there immediately.

mtmorgan commented on vignettes/BiocCheck.Rmd:
```diff
@@ -550,10 +550,48 @@ The output of this check includes the first 6 offending lines of code;
+verify that your code runs as intended after you re-format your code.
 
 [formatR]: https://cran.r-project.org/web/packages/formatR/index.html
+[styler]: https://styler.r-lib.org/
```
CRAN reference, please

mtmorgan reviewed (commented):
This also looks promising, but the use of biocthis is problematic at the moment.

Also, do 'we' really have enough experience to know that styler is definitely 'better' than formatR? I don't know, personally... What are the particular features that make it stand out?

lcolladotor commented:
Hi,

Thank you for your PR review =)

* I'll remove the `biocthis` mentions as requested. I had asked before though, but well, I understand now why you don't want it mentioned thanks to your comments above.
* I'll switch to CRAN links also.
* Regarding `Also, do 'we' really have enough experience to know that styler is definitely 'better' than formatR? I don't know, personally... What are the particular features that make it stand out?`. I can tell you that `formatR` broke my code (if you have code that requires both `"` and `'` to run, `formatR` changes the single quotes to double and that can break the code), doesn't update `roxygen2` example sections (which means that you still have a lot of manual work to do), and is less actively maintained (given the push at `r-lib` towards `styler`). `formatR` does attempt to deal with long lines (unlike `styler` given https://github.com/r-lib/styler/issues/247). The proposed vignette changes mention this while having the `handleMessage()` prompt refer to `styler` and the `BiocCheck` vignette. So `formatR` will still be mentioned as a possible solution.

Best,
Leo

lcolladotor committed:
Respond to the PR review

This commit does the things outlined at https://github.com/Bioconductor/BiocCheck/pull/88#issuecomment-630921073. That is:

* Removes any `biocthis` mentions.
* Uses CRAN links for `styler`.
* Fixes a typo.
* No longer bumps the version number to avoid merge conflicts with other BiocCheck-a-thon pull requests as suggested on Slack.

lcolladotor commented:
Hi,

I just sent a new commit to this PR in response to the review comments.

Best,
Leo
